src/assign.rs

```diff
@@ -1,48 +1,55 @@
-use alloc::borrow::Cow;
-use serde_json::Value;
+use core::mem::replace;
 
-use crate::{Error, Pointer, PointerBuf};
+use serde_json::{map::Entry, Map, Value};
+use snafu::ResultExt;
+
+use crate::{assign_error, AssignError, Pointer, Token};
```

Comment:
We can have a separate discussion about this, but worth converging on a style for imports. I generally prefer to just merge everything into a single block, have never found it useful to separate them, but if you feel strongly against it, I can accept this convention too.

Reply:
Oh, I have no preference here. I'm lazy and let RA just do its thing. I'm definitely game for cleaning them up. I think there's even a lint that'll do it but its in nightly and was been hit-or-miss for me when I tried it last.

Reply:
I think you're right. It's a bit unfortunate RA doesn't do exactly what I want by default, and there's no setting for it, but if you just merge the blocks once it'll keep them merged (even if you add more imports later). No need to consolidate this everywhere in this PR, but might as well start with it for new code.

Reply:
I think the untidiness of imports is amplified by all the unnecessary mods in the crate. Flattening it should make this a lot easier to keep in check. 

I may try that lint again too. I was super excited when I found it because while I do find the utility of auto-imports super convenient, I do get annoyed and distracted by cluttered use blocks.